app/uk/gov/hmrc/apisubscriptionfields/model/Model.scala

```diff
@@ -28,6 +29,16 @@ case class ApiVersion(value: String) extends AnyVal
 
 case class SubscriptionFieldsId(value: UUID) extends AnyVal
 
+object ValidationRuleType extends Enumeration {
+  type ValidationRuleType = Value
+
+  val REGEX = Value("REGEX")
+}
+
+case class ValidationRule(validationRuleType: ValidationRuleType, value: String)
```

Comment:
This won't allow any extension.  We probably need something with a 
`sealed trait ValidationRule
case class RegexValidationRule(regex: String) extends ValidationRule
`

and then use [play-json-derived](https://github.com/julienrf/play-json-derived-codecs/tree/v3.3) to provide read/write of the appropriate types (it uses the class name as discriminator)

The rest remains almost as it is.